backend_design_doc.md

```diff
@@ -0,0 +1,307 @@
+<p><b>*NOTE:</b> "schedules" as used in endpoints refer to <b>childrens'</b> <b>enrollments</b> to <b>courses</b>; "schedule" specific endpoints join the <b>children</b>, <b>enrollments</b>, and <b>courses</b> tables and return data matching the given <b>profile_id</b></p>
+
+<p> *NOTE: A JWT will be included in the authorization header for every request <p>
+
+# Profiles
+
+```
+
+{
+  profile_id: INCREMENT (primary key, auto-increments, generated by database),
+  name: STRING (required),
+  email: STRING (required),
+  role_id: INTEGER (required, foreign key),
+  avatarUrl: STRING (required, defaults to: 'https://i.stack.imgur.com/frlIf.png'),
+}
+```
+
+| Method   | URL                               | Description                                                                                                              |
+| -------- | --------------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
+| [GET]    | /profile/                         | Returns an array of all existing profiles.                                                                               |
+| [GET]    | /profile/{profile_id}             | Returns an array filled with event objects that contains information based on profile_id and role_id.                    |
+| [GET]    | /profile/role/{role_id}           | Returns an array filled with event objects that contain information based on role_id for all profiles of a role_id type. |
+| [POST]   | /profile/                         | Requires a name, and email. Registers a new user.                                                                        |
+| [PUT]    | /profile/{profile_id}             | Updates specific profile.                                                                                                |
+
+#### User:
+
+<p>These endpoints are user-focused. As opposed to the more flexible Profile endpoints where <b>profile_id</b> must be specified, these endpoints retrieve data specific only to the user profile that is making the API request by using the logged-in user's <b>profile_id</b>.</p>
+
+| Method | URL              | Description                                                                 |
+| ------ | ---------------- | --------------------------------------------------------------------------- |
+| [GET]  | /user/           | Returns an event object with the specified `type`.                          |
+| [GET]  | /user/inbox/     | Returns an event object                                                     |
```

Comment:
Maybe add periods at the end of the sentences on line 33 and 34 to keep with the pattern.